File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/javascript/reverse_string.js
function reverseString(str) {
  //create empty str
  let reversedStr = ""

  //iterate backwards through the given string 
  //then add to new string
  for (let i = str.length - 1; i >= 0; i--) {
    reversedStr += str[i]
  }
  return reversedStr
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  # split string
  split_str = str.split("")
  # create empty array
  reversed = []
  # iterate through string and shovel into array
  str.size.times { reversed << split_str.pop }
  # join reversed string and return
  return reversed.join
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution